Index: jetson_multimedia_api/samples/Rules.mk
===================================================================
--- jetson_multimedia_api.orig/samples/Rules.mk
+++ jetson_multimedia_api/samples/Rules.mk
@@ -28,10 +28,6 @@
 #
 ###############################################################################
 
-# Clear the flags from env
-CPPFLAGS :=
-LDFLAGS :=
-
 # Verbose flag
 ifeq ($(VERBOSE), 1)
 AT =
@@ -39,11 +35,6 @@ else
 AT = @
 endif
 
-# ARM ABI of the target platform
-ifeq ($(TEGRA_ARMABI),)
-TEGRA_ARMABI ?= aarch64-linux-gnu
-endif
-
 # Location of the target rootfs
 ifeq ($(shell uname -m), aarch64)
 TARGET_ROOTFS :=
@@ -62,49 +53,16 @@ CLASS_DIR 	:= $(TOP_DIR)/samples/common/
 ALGO_CUDA_DIR 	:= $(TOP_DIR)/samples/common/algorithm/cuda
 ALGO_TRT_DIR 	:= $(TOP_DIR)/samples/common/algorithm/trt
 
-ifeq ($(shell uname -m), aarch64)
-CROSS_COMPILE :=
-else
-CROSS_COMPILE ?= aarch64-unknown-linux-gnu-
-endif
-AS             = $(AT) $(CROSS_COMPILE)as
-LD             = $(AT) $(CROSS_COMPILE)ld
-CC             = $(AT) $(CROSS_COMPILE)gcc
-CPP            = $(AT) $(CROSS_COMPILE)g++
-AR             = $(AT) $(CROSS_COMPILE)ar
-NM             = $(AT) $(CROSS_COMPILE)nm
-STRIP          = $(AT) $(CROSS_COMPILE)strip
-OBJCOPY        = $(AT) $(CROSS_COMPILE)objcopy
-OBJDUMP        = $(AT) $(CROSS_COMPILE)objdump
-NVCC           = $(AT) $(CUDA_PATH)/bin/nvcc -ccbin $(filter-out $(AT), $(CPP))
-
-# Specify the logical root directory for headers and libraries.
-ifneq ($(TARGET_ROOTFS),)
-CPPFLAGS += --sysroot=$(TARGET_ROOTFS)
-LDFLAGS += \
-	-Wl,-rpath-link=$(TARGET_ROOTFS)/lib/$(TEGRA_ARMABI) \
-	-Wl,-rpath-link=$(TARGET_ROOTFS)/usr/lib/$(TEGRA_ARMABI) \
-	-Wl,-rpath-link=$(TARGET_ROOTFS)/usr/lib/$(TEGRA_ARMABI)/tegra \
-	-Wl,-rpath-link=$(TARGET_ROOTFS)/$(CUDA_PATH)/lib64
-endif
-
 # All common header files
 CPPFLAGS += -std=c++11 \
 	-I"$(TOP_DIR)/include" \
 	-I"$(TOP_DIR)/include/libjpeg-8b" \
 	-I"$(ALGO_CUDA_DIR)" \
-	-I"$(ALGO_TRT_DIR)" \
-	-I"$(TARGET_ROOTFS)/$(CUDA_PATH)/include" \
-	-I"$(TARGET_ROOTFS)/usr/include/$(TEGRA_ARMABI)" \
-	-I"$(TARGET_ROOTFS)/usr/include/libdrm" \
-	-I"$(TARGET_ROOTFS)/usr/include/opencv4"
+	-I"$(ALGO_TRT_DIR)"
 
 # All common dependent libraries
 LDFLAGS += \
 	-lpthread -lv4l2 -lEGL -lGLESv2 -lX11 \
 	-lnvbuf_utils -lnvjpeg -lnvosd -ldrm \
 	-lcuda -lcudart \
-	-lnvinfer -lnvparsers \
-	-L"$(TARGET_ROOTFS)/$(CUDA_PATH)/lib64" \
-	-L"$(TARGET_ROOTFS)/usr/lib/$(TEGRA_ARMABI)" \
-	-L"$(TARGET_ROOTFS)/usr/lib/$(TEGRA_ARMABI)/tegra"
+	-lnvinfer -lnvparsers
Index: jetson_multimedia_api/samples/v4l2cuda/Makefile
===================================================================
--- jetson_multimedia_api.orig/samples/v4l2cuda/Makefile
+++ jetson_multimedia_api/samples/v4l2cuda/Makefile
@@ -26,14 +26,8 @@ SRCS := \
 	capture.cpp \
 	yuv2rgb.cu
 
-ALL_CPPFLAGS := $(addprefix -Xcompiler ,$(filter-out -std=c++11, $(CPPFLAGS)))
-
-# CUDA code generation flags
-GENCODE_SM53 := -gencode arch=compute_53,code=sm_53
-GENCODE_SM62 := -gencode arch=compute_62,code=sm_62
-GENCODE_SM72 := -gencode arch=compute_72,code=sm_72
-GENCODE_SM_PTX := -gencode arch=compute_72,code=compute_72
-GENCODE_FLAGS := $(GENCODE_SM53) $(GENCODE_SM62) $(GENCODE_SM72) $(GENCODE_SM_PTX)
+ALL_CPPFLAGS := $(NVCCFLAGS)
+ALL_CPPFLAGS += $(addprefix -Xcompiler ,$(filter-out -std=c++11, $(CPPFLAGS)))
 
 all: $(APP)
 
@@ -47,7 +41,7 @@ yuv2rgb.o: yuv2rgb.cu
 
 $(APP): capture.o yuv2rgb.o
 	@echo "Linking: $@"
-	$(CPP) -o $@ $^ $(CPPFLAGS) $(LDFLAGS)
+	$(CPP) -o $@ $^ $(CPPFLAGS) $(LDFLAGS) $(NVLDFLAGS)
 
 clean:
 	$(AT) rm -f *.o $(APP)
Index: jetson_multimedia_api/samples/common/algorithm/cuda/Makefile
===================================================================
--- jetson_multimedia_api.orig/samples/common/algorithm/cuda/Makefile
+++ jetson_multimedia_api/samples/common/algorithm/cuda/Makefile
@@ -43,13 +43,6 @@ endif
 ALL_CPPFLAGS := $(NVCCFLAGS)
 ALL_CPPFLAGS += $(addprefix -Xcompiler ,$(filter-out -std=c++11, $(CPPFLAGS)))
 
-# CUDA code generation flags
-GENCODE_SM53 := -gencode arch=compute_53,code=sm_53
-GENCODE_SM62 := -gencode arch=compute_62,code=sm_62
-GENCODE_SM72 := -gencode arch=compute_72,code=sm_72
-GENCODE_SM_PTX := -gencode arch=compute_72,code=compute_72
-GENCODE_FLAGS := $(GENCODE_SM53) $(GENCODE_SM62) $(GENCODE_SM72) $(GENCODE_SM_PTX)
-
 # Target rules
 all: NvAnalysis.o NvCudaProc.o
 
