Index: tegra_multimedia_api/tools/ConvertCaffeToTrtModel/Makefile
===================================================================
--- tegra_multimedia_api.orig/tools/ConvertCaffeToTrtModel/Makefile
+++ tegra_multimedia_api/tools/ConvertCaffeToTrtModel/Makefile
@@ -28,16 +28,8 @@
 #
 ###############################################################################
 
-CC = aarch64-linux-gnu-g++
-
-INCPATHS = -I/usr/include/aarch64-linux-gnu/
-CFLAGS = -Wall -std=c++11 $(INCPATHS)
-
-CFLAGS += -D ENABLE_TRT
-
-LIBPATHS = -L/usr/lib/aarch64-linux-gnu/
+CXXFLAGS += -Wall -std=c++11 -DENABLE_TRT
 LIBS = -lnvinfer -lnvcaffe_parser
-LFLAGS = $(LIBPATHS) -Wl,--start-group $(LIBS) -Wl,--end-group
 
 SRCFILES = ConvertCaffeToTrtModel_main.cpp
 OUTNAME = ConvertCaffeToTrtModel
@@ -45,7 +37,7 @@ OUTNAME = ConvertCaffeToTrtModel
 all: $(OUTNAME)
 
 $(OUTNAME) : $(SRCFILES)
-	$(CC) -o $@ $^ $(CFLAGS) $(LFLAGS)
+	$(CXX) -o $@ $^ $(CXXFLAGS) -Wl,--start-group $(LIBS) -Wl,--end-group $(LDFLAGS)
 
 clean:
 	rm -rf $(OUTNAME)
