Index: tegra_multimedia_api/tools/ConvertCaffeToTrtModel/ConvertCaffeToTrtModel_main.cpp
===================================================================
--- tegra_multimedia_api.orig/tools/ConvertCaffeToTrtModel/ConvertCaffeToTrtModel_main.cpp
+++ tegra_multimedia_api/tools/ConvertCaffeToTrtModel/ConvertCaffeToTrtModel_main.cpp
@@ -31,6 +31,7 @@
 #include <iostream>
 #include <cmath>
 #include <algorithm>
+#include <vector>
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <time.h>
